pkg/apis/v1/nodepool_conversion_test.go

```diff
@@ -269,6 +270,24 @@ var _ = Describe("Convert V1 to V1beta1 NodePool API", func() {
 			Expect(v1beta1nodepool.Status.Resources[resource]).To(Equal(v1nodepool.Status.Resources[resource]))
 		}
 	})
+	Context("Round Trip", func() {
+		DescribeTable(
+			"NillableDuration",
+			func(value string, field func() *NillableDuration) {
+				str := fmt.Sprintf("%q", value)
+				duration := NillableDuration{}
+				Expect(json.Unmarshal([]byte(str), &duration)).Should(Succeed())
+				*field() = duration
+				Expect(v1nodepool.ConvertTo(ctx, v1beta1nodepool)).To(Succeed())
```

Comment:
Just validating: This test failed prior to this change?

Reply:
Also, any reason to not json marshal the entire object rather than just the specific field that we care about during the test? Seems like a stronger bar to hold if we do that

Reply:
Yep, this would have failed before. I was only marshaling this as json to ensure we populated the "raw" value correctly before doing the conversions, I've updated every instance of us creating a NillableDuration with a constructor that covers this instead.